src/fastoad/openmdao/variables.py

```diff
@@ -618,15 +635,15 @@ def from_problem(
                         # We already have a non-NaN value and current variable provides no unit.
                         # No need for using the current variable.
                         continue
-
-                promoted_outputs[prom_name] = metadata
-
-            # Remove from inputs the variables that are outputs of some other component
-            promoted_inputs = {
-                metadata["prom_name"]: dict(metadata, is_input=True)
-                for metadata in inputs.values()
-                if metadata["prom_name"] not in promoted_outputs
-            }
+                if prom_name not in promoted_inputs:
+                    promoted_outputs[prom_name] = metadata
+
+            # # Remove from inputs the variables that are outputs of some other component
+            # promoted_inputs = {
+            #     metadata["prom_name"]: dict(metadata, is_input=True)
+            #     for metadata in inputs.values()
+            #     if metadata["prom_name"] not in promoted_outputs
+            # }
```

Comment:
I missed that in my previous reading. Could you please remove these lines ?

Reply:
Thanks. Removed in fbe8bbb.